.github/workflows/build_sdist.yml

```diff
@@ -0,0 +1,96 @@
+name: Build and test sdist
+
+on:
+  workflow_dispatch:
+    inputs:
+      ref:
+        description: "Target branch / tag / commit"
+        required: true
+        type: string
+  workflow_call:
+    inputs:
+      ref:
+        description: "Target branch / tag / commit"
+        required: true
+        type: string
+
+jobs:
+  build-sdist:
+    name: Build sdist
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.ref }}
+          fetch-depth: 0
+          fetch-tags: true
+
+      - name: Create source distribution
+        shell: bash -l {0}
+        run: |
+          pipx run build --sdist .
+
+      - name: Upload source dist as build artifact
+        uses: actions/upload-artifact@v4
+        with:
+            name: python-source-distribution
+            path: dist/
+            if-no-files-found: error
+
+  test-sdist:
+    needs: build-sdist
+    name: Test build from sdist on Windows
```

Comment:
Is there a test on mac/Linux for the sdist? Am I missing it?

Reply:
Nah, just testing it on one platform. Windows was the platform that had trouble in the past which is why it was used; I just ported it here from previous workflow.

Can probably switch back to ubuntu now if we like, and save a little CI time/budget.